src/components/SpaceCreateVoting.vue

```diff
@@ -53,19 +54,55 @@ watch(
   { immediate: true }
 );
 
+// we need to watch for selection change to properly update the voting form stae
+watch(
+  () => props.osnap.selection,
+  () => {
+    // If using osnap, we can only allow basic voting, for, against, abstain
+    if (props.osnap.selection) {
+      form.value.type = 'basic';
+    } else {
+      // Initialize the proposal type if set in space
+      if (props.space?.voting?.type) form.value.type = props.space.voting.type;
+    }
+  }
+);
+
 const { getSnapshot } = useSnapshot();
 
 onMounted(async () => {
   // Initialize the start date to current
   if (!sourceProposalLoaded.value && !userSelectedDateStart.value)
     form.value.start = Number((Date.now() / 1e3).toFixed());
-  // Initialize the proposal type if set in space
-  if (props.space?.voting?.type) form.value.type = props.space.voting.type;
   form.value.snapshot = await getSnapshot(props.space.network);
 });
+
+defineEmits<{
+  (event: 'osnapToggle'): void;
+}>();
 </script>
 
 <template>
+  <div v-if="osnap.enabled">
+    <h6>OSnap Proposal</h6>
+    <p>
+      Are you planning for this proposal to initiate a transaction that your
+      organizations safe will execute if approved? (Remember, oSnap enables
+      trustless and permissionless execution):
+    </p>
+    <br />
+    <input
+      id="toggleOsnap"
+      type="checkbox"
+      :checked="osnap.selection"
+      @change="$emit('osnapToggle')"
+    />
+    <label for="toggleOsnap">
+      Yes, use Osnap for transactions ( this will restrict voting types ).
```

Comment:
We should use the official casing here
```suggestion
      Yes, use oSnap for transactions (this will restrict voting types).
```